src/components/edit-dl-controller.tsx

```diff
@@ -0,0 +1,141 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import React, { FC, useCallback, useEffect, useMemo, useRef, useState } from 'react';
+
+import {
+	Divider,
+	ModalBody,
+	ModalFooter,
+	ModalHeader,
+	useSnackbar
+} from '@zextras/carbonio-design-system';
+import { difference, xor } from 'lodash';
+import { useTranslation } from 'react-i18next';
+
+import { EditDLComponent } from './edit-dl';
+import { client } from '../network/client';
+
+export type EditDLControllerComponentProps = {
+	email: string;
+	displayName: string;
+	onClose: () => void;
+	onSave: () => void;
+};
+
+export const getMembersPartition = (
+	originalMembers: Array<string>,
+	updatedMembers: Array<string>
+): { membersToAdd: Array<string>; membersToRemove: Array<string> } => ({
+	membersToAdd: difference(updatedMembers, originalMembers),
+	membersToRemove: difference(originalMembers, updatedMembers)
+});
+
+export const EditDLControllerComponent: FC<EditDLControllerComponentProps> = ({
+	email,
+	displayName,
+	onClose,
+	onSave
+}) => {
+	const [members, setMembers] = useState<string[]>([]);
+	const originalMembersRef = useRef<string[]>([]);
+	const [totalMembers, setTotalMembers] = useState<number>(0);
```

Comment:
Probably this state is redundant, totalMembers must be always computed as members.length;
Have a derivative state to update could be error prone

Reply:
I don't agree entirely. At the moment it is, but we have the total number of members specified as a different value, set from the response of the backend, because we could have pagination for the members (and probably we should one day). By doing so, we are sure this state is the “really total” count, while the length of the members could be only a partial subset of the full list.